libc/string_lenchr.c

```diff
@@ -150,9 +169,259 @@ TEST(string_len, big)
 }
 
 
+/*
+////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_chr)
+{
+}
+
+
+TEST_TEAR_DOWN(string_chr)
+{
+}
+
+
+TEST(string_chr, basic_use)
+{
+
+	const char *lorem = "Lorem ipsum dolor sit amet, consectetur adipiscing elit y";
+	int sz;
+
+	sz = strlen(lorem);
+
+	/* Check of proper working base */
+	TEST_ASSERT_EQUAL_STRING(lorem, strchr(lorem, 'L'));
+	TEST_ASSERT_EQUAL_STRING(lorem, strrchr(lorem, 'L'));
+	TEST_ASSERT_EQUAL_STRING(lorem, memchr(lorem, 'L', sz));
+
+	TEST_ASSERT_EQUAL_STRING("lor sit amet, consectetur adipiscing elit y", strchr(lorem, 'l'));
+	TEST_ASSERT_EQUAL_STRING("lit y", strrchr(lorem, 'l'));
+	TEST_ASSERT_EQUAL_STRING("lor sit amet, consectetur adipiscing elit y", memchr(lorem, 'l', sz));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr(lorem, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr(lorem, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(lorem, 'x', sz));
+
+	TEST_ASSERT_EQUAL_STRING(&lorem[sz], strchr(lorem, lorem[sz]));
+	TEST_ASSERT_EQUAL_STRING(&lorem[sz], strrchr(lorem, lorem[sz]));
+	TEST_ASSERT_EQUAL_STRING(&lorem[sz], memchr(lorem, lorem[sz], sz + 1));
+}
+
+
+TEST(string_chr, big)
+{
+	char str[PATH_MAX] = { 0 };
+
+	/* Long string case */
+	memset(str, 'a', PATH_MAX - 1);
+	str[PATH_MAX - 5] = 'b';
+
+	TEST_ASSERT_EQUAL_STRING("baaa", strchr(str, 'b'));
+	TEST_ASSERT_EQUAL_STRING("baaa", strrchr(str, 'b'));
+	TEST_ASSERT_EQUAL_STRING("baaa", memchr(str, 'b', sizeof(str)));
```

Comment:
You can add there checks similar to the ones from the previous test cases, so above that you can add checking:

- the termination character
- the first character
- the character, which is not present in the string